src/app/resolver/country.resolver.ts

```diff
@@ -18,7 +18,7 @@ export class CountryResolver implements Resolve<CountryDetail | null> {
     if (countryName) {
       countryName = countryName.replace(/-/g, ' '); // Convert hyphen back to space
 
-      return this.countryService.getCountryByName(countryName);
+      return this.countryService.getAllCountries();
```

Comment:
Can you call the action to loadCoutries here, calling services inside a resolver isn't a best practice